hibernate-core/src/main/java/org/hibernate/query/sqm/internal/QuerySqmImpl.java

```diff
@@ -933,7 +933,7 @@ private NonSelectQueryPlan buildInsertQueryPlan() {
 		final EntityPersister entityDescriptor = getSessionFactory().getDomainModel().findEntityDescriptor( entityNameToInsert );
 
 		final SqmMultiTableInsertStrategy multiTableStrategy = entityDescriptor.getSqmMultiTableInsertStrategy();
-		if ( multiTableStrategy == null ) {
+		if ( multiTableStrategy == null || sqmInsert instanceof SqmInsertValuesStatement ) {
```

Comment:
Hmm, was that an issue somehow for you because I don't know if it is safe to simply do this, given that a VALUES clause can have expressions.

Reply:
Yes, I was fixing the fact that `insert ... values ...` was broken for the _very first example I tried_.

Reply:
(Specifically, it was broken, because it was trying to insert `values()` into a temp table, and didn't assign a value to the `rn_` column.)

Reply:
Ah, that's a bug :)

I guess your tests will reproduce that, so I'll take a look and see if I can change this logic somehow.

Reply:
Yup, they will.